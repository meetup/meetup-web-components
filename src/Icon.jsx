import React from 'react';
import cx from 'classnames';

export const ICON_CLASS = 'svg';

export const MEDIA_SIZES = {
	xs: '16',
	s: '24',
	m: '36',
	l: '48',
	xl: '72',
};

const spritePath = require('file!swarm-icons/dist/sprite/sprite.inc');

/**
 * Icon component used to insert an svg icon into a component or page
 *
 * **Accessibility** If an Icon is used on its own without supporting
 * text to explain what it is/does, be a good citizen and pass in an
 * `aria-label` attribute describing what the icon represents
 *
 * @module Icon
 */
class Icon extends React.Component {

	render() {
		const {
			className,
			shape,
			size,
			...other
		} = this.props;

		const classNames = cx(
			ICON_CLASS,
			`svg--${shape}`,
			className
		);

		const dim = MEDIA_SIZES[size];

		return (
			<span className={classNames}>
				<svg
					preserveAspectRatio='xMinYMin meet'
					width={dim}
					height={dim}
					viewBox={`0 0 ${dim} ${dim}`}
					className='svg-icon valign--middle'
					role='img'
					aria-labelledby={`${shape}_title`}
					{...other}>
					<title id={`${shape}_title`} />
					<use role='presentation' xlinkHref={`${spritePath}#icon-${shape}`} />
				</svg>
			</span>
		);
	}
}

Icon.defaultProps = {
	size: 'm'
};

Icon.propTypes = {
	shape: React.PropTypes.string.isRequired,
	size: React.PropTypes.oneOf(Object.keys(MEDIA_SIZES))
};

export default Icon;
