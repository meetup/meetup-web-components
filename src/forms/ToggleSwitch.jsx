import PropTypes from 'prop-types';
import React from 'react';

import { Toggle } from '@meetup/swarm-components';

/**
 * @module ToggleSwitch
 */
class ToggleSwitch extends React.Component {
	render() {
		const { isActive, label, labelledBy, id, name, ...other } = this.props;

		const labelProps = labelledBy ? { 'aria-labelledby': labelledBy } : {};

		return (
			<div>
				{label && <label htmlFor={id || name}>{label}</label>}
				<Toggle
					id={id || name}
					name={name}
					checked={isActive}
					{...labelProps}
					{...other}
				/>
			</div>
		);
	}
}

ToggleSwitch.propTypes = {
	/** Whether to use disabled attribute on the button element and apply disabled switch styles */
	disabled: PropTypes.bool,

	/** Whether the switch is toggled on */
	isActive: PropTypes.bool,

	/** What we render into the input's `<label />` */
	label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

	/** The id of the element that labels the toggle switch */
	labelledBy: PropTypes.string,

	/** The identifier used to associate a label with a toggle switch */
	name: PropTypes.string.isRequired,
};

export default ToggleSwitch;
