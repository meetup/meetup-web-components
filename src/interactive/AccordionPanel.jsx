import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';

import Chunk from '../layout/Chunk';
import Flex from '../layout/Flex';
import FlexItem from '../layout/FlexItem';
import Icon from '../media/Icon';
import ToggleSwitch from '../forms/ToggleSwitch';
import LockedBadge from '../forms/LockedBadge';

export const PANEL_CLASS = 'accordionPanel';
export const ACTIVEPANEL_CLASS = 'accordionPanel--active';
export const ACCORDION_LABEL_CLASS = 'accordionPanel-label';

/**
 * @module AccordionPanel
 */
class AccordionPanel extends React.Component {
	constructor(props) {
		super(props);
		this._handleToggle = this._handleToggle.bind(this);
		this.onKeyUp = this.onKeyUp.bind(this);
		this.onToggleClick = this.onToggleClick.bind(this);
		this.onTransitionEnd = this.onTransitionEnd.bind(this);
		this.handleLockedLabelClick = this.handleLockedLabelClick.bind(this);

		this.state = {
			height: props.isOpen ? 'auto' : '0px',
		};
	}

	/**
	 * @parm {Boolean} isOpen
	 * @parm {Ref} contentEl - element ref
	 * @returns {Object} inline styles
	 */
	getPanelStyle(isOpen, contentEl) {
		const style = { height: '0px' };

		// set height to 0
		if (!isOpen || !contentEl) {
			return style;
		}

		// set height to {n}px
		style.height = `${contentEl.getBoundingClientRect().height}px`;
		return style;
	}

	/**
	 *
	 * @description calls the AccordionPanelGroups's callback to toggle open state
	 * and render the `AccordionPanel` open or closed, sets height in state
	 * @param {Event} e - the event object
	 * @returns {undefined}
	 */
	_handleToggle(e) {
		e.preventDefault();

		const {
			isOpen,
			setClickedPanel,
			onClickCallback,
			panelIndex,
			disableAndOpen,
			isDisabledPanelOpen,
			isLocked,
		} = this.props;
		if (isLocked) return;
		if (disableAndOpen) {
			setClickedPanel &&
				setClickedPanel(e, {
					panelIndex: panelIndex,
					isDisabledPanelOpen: true,
				});
			return this.getPanelStyle(isDisabledPanelOpen, this.contentEl);
		}
		setClickedPanel &&
			setClickedPanel(e, {
				panelIndex: panelIndex,
				isOpen: !isOpen,
			});
		this.getPanelStyle(!isOpen, this.contentEl);
		onClickCallback && onClickCallback(e, !isOpen);
	}

	onToggleClick(e) {
		e.preventDefault();

		this.props.onToggleClick ? this.props.onToggleClick(e) : this._handleToggle(e);
	}

	/**
	 * @description allows the AccordionPanel content to be toggled with the "Enter" and "space" keys
	 * @returns {undefined}
	 */
	onKeyUp(e) {
		const isActivatingButton = [' ', 'Enter'].some(key => e.key === key);

		if (isActivatingButton) {
			this._handleToggle(e);
		}
	}

	/**
	 * @description forceUpdate allows us to calculate height again now that contentEl is set
	 * @returns {undefined}
	 */
	componentDidMount() {
		this.forceUpdate();
	}

	/**
	 * @description set height state when the panel receives a change
	 * @returns {undefined}
	 */
	UNSAFE_componentWillReceiveProps(nextProps) {
		if (nextProps.isOpen !== this.props.isOpen) {
			this.setState(
				() => this.getPanelStyle(!nextProps.isOpen, this.contentEl),
				() => {
					setTimeout(() => {
						this.setState(() =>
							this.getPanelStyle(nextProps.isOpen, this.contentEl)
						);
					}, 1);
				}
			);
		}
	}

	/**
	 * @returns {String} icon shape
	 */
	getIconShape() {
		const { indicatorIcon, indicatorIconActive } = this.props;

		return this.props.isOpen && indicatorIconActive
			? indicatorIconActive
			: indicatorIcon;
	}

	/**
	 * @returns undefined
	 */
	onTransitionEnd() {
		if (this.props.isOpen) {
			this.setState(() => ({
				height: 'auto',
			}));
		}
	}

	handleLockedLabelClick(e) {
		e.preventDefault();

		this.props.onLockedLabelClick(e);
	}

	render() {
		const {
			panelContent,
			panelIndex,
			label,
			isOpen,
			setClickedPanel, // eslint-disable-line no-unused-vars
			onClickCallback, // eslint-disable-line no-unused-vars
			indicatorAlign, // eslint-disable-line no-unused-vars
			indicatorIcon, // eslint-disable-line no-unused-vars
			indicatorIconActive, // eslint-disable-line no-unused-vars
			indicatorIconSize,
			indicatorSwitch,
			classNamesActive,
			className,
			onToggleClick, // eslint-disable-line no-unused-vars
			disableAndOpen,
			isDisabledPanelOpen, // eslint-disable-line no-unused-vars
			isLocked,
			lockedLabel,
			onLockedLabelClick, // eslint-disable-line no-unused-vars
			...other
		} = this.props;

		const classNames = {
			accordionPanel: cx(
				'padding--bottom padding--top',
				PANEL_CLASS,
				{
					[ACTIVEPANEL_CLASS]: isOpen,
					[classNamesActive]: isOpen && classNamesActive,
				},
				className
			),
			trigger: cx(className, ACCORDION_LABEL_CLASS, 'display--block span--100'),
			content: cx('accordionPanel-animator', {
				'accordionPanel-animator--collapse': !isOpen,
			}),
			icon: cx('indicator', {
				'indicator--animateActive':
					this.getIconShape() == 'chevron-down' && isOpen,
			}),
		};

		const isLabelString = label && typeof label == 'string';
		const panelId = isLabelString
			? label.replace(/\s+/g, '').toLowerCase()
			: panelIndex;

		return (
			<div>
				<div
					role="tab"
					aria-controls={`panel-${panelId}`}
					aria-expanded={isOpen}
					aria-selected={isOpen}
					className={classNames.trigger}
					tabIndex={0}
					onKeyUp={this.onKeyUp}
					onClick={this.onToggleClick}
				>
					<Flex
						className={classNames.accordionPanel}
						rowReverse={indicatorAlign === 'left' && 'all'}
						tabIndex={-1}
						align="center"
						{...other}
					>
						{isLocked ? (
							<Flex
								wrap
								direction="column"
								className="accordionPanel-locked"
							>
								<FlexItem
									id={`label-${panelId}`}
									className="accordionPanel-locked-panel-label"
									shrink
								>
									{label}
								</FlexItem>
								<LockedBadge
									label={lockedLabel}
									onClick={onLockedLabelClick}
								/>
							</Flex>
						) : (
							<FlexItem id={`label-${panelId}`}>{label}</FlexItem>
						)}

						<FlexItem
							className="accordionPanel-icon"
							onClick={this.onToggleClick}
							shrink
						>
							{!indicatorSwitch && indicatorIcon ? (
								<Icon
									shape={this.getIconShape()}
									size={indicatorIconSize}
									className={classNames.icon}
								/>
							) : (
								<ToggleSwitch
									tabIndex="-1"
									isActive={isOpen && !disableAndOpen}
									disabled={!!onToggleClick || !!disableAndOpen}
									id={`switch-${panelId}`}
									labelledBy={`label-${panelId}`}
									name={panelId}
									onClick={this.onToggleClick}
								/>
							)}
						</FlexItem>
					</Flex>
				</div>

				<Chunk
					role="tabpanel"
					id={`panel-${panelId}`}
					aria-labelledby={`label-${panelId}`}
					aria-hidden={!isOpen}
					className={classNames.content}
					style={{ height: this.state.height }}
					onTransitionEnd={this.onTransitionEnd}
				>
					<div
						className={'accordionPanel-content'}
						ref={div => {
							this.contentEl = div;
						}}
					>
						{panelContent}
					</div>
				</Chunk>
			</div>
		);
	}
}

AccordionPanel.defaultProps = {
	isOpen: false,
	indicatorAlign: 'right',
	indicatorIcon: 'chevron-down',
	indicatorIconSize: 'xs',
};

AccordionPanel.propTypes = {
	/** zero-indexed position of the panel */
	panelIndex: PropTypes.number,

	/** Class names to add to the panel wrapper when the panel is open */
	classNamesActive: PropTypes.string,

	/** Whether the panel is open */
	isOpen: PropTypes.oneOf([null, true, false]),

	/** The content to render inside the detail area of an expanded panel */
	panelContent: PropTypes.element,

	/** A function used to set the clicked panel to it's opened state */
	setClickedPanel: PropTypes.func,

	/** A callback that happens after the panel has been toggled */
	onClickCallback: PropTypes.func,

	/** Content to render in the part of the panel that is always visible (does not include indicator icon or switch) */
	label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,

	/** Class names to add to the wrapper of the panel label and indicator */
	className: PropTypes.string,

	/** Which side of each panel that the icon or indicator switch renders on */
	indicatorAlign: PropTypes.string,

	/** Shape name of the icon to render as an indicator for closed panels */
	indicatorIcon: PropTypes.string,

	/** Shape name of the icon to render as an indicator for opened panels */
	indicatorIconActive: PropTypes.string,

	/** Size key for the icon rendered for open and closed indicators */
	indicatorIconSize: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl']),

	/** Whether to use a ToggleSwitch as an indicator instead of an icon */
	indicatorSwitch: PropTypes.bool,

	/** Prevent ToggleSwitch active but show content after click  */
	disableAndOpen: PropTypes.bool,

	/** Whether the panel is locked due to some conditional requirements */
	isLocked: PropTypes.bool,

	/** Label of CTA badge for unlocking the panel */
	lockedLabel: PropTypes.string,

	/** A callback that happens after the locked label has been clicked  */
	onLockedLabelClick: PropTypes.func,
};

export default AccordionPanel;
