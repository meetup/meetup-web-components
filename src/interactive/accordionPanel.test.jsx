import React from 'react';
import { shallow, mount } from 'enzyme';

import AccordionPanel, { ACTIVEPANEL_CLASS } from './AccordionPanel';
import Icon from '../media/Icon';
import ToggleSwitch from '../forms/ToggleSwitch';
import { textContent1 } from '../../__mocks__/textContentMocks';

describe('AccordionPanel', function() {
	let panel, openPanel;

	const customIcon = 'plus';
	const customIconActive = 'minus';

	let handlePanelClickCallback, onClickCallback;

	beforeEach(() => {
		handlePanelClickCallback = jest.fn();
		onClickCallback = jest.fn();

		panel = shallow(
			<AccordionPanel
				label="First Section"
				setClickedPanel={handlePanelClickCallback}
				onClickCallback={onClickCallback}
				panelContent={
					<div className="runningText">
						<p>{textContent1}</p>
					</div>
				}
			/>
		);

		openPanel = shallow(
			<AccordionPanel
				isOpen
				label="First Section"
				panelContent={
					<div className="runningText">
						<p>{textContent1}</p>
					</div>
				}
			/>
		);
	});

	afterEach(() => {
		panel = null;
		openPanel = null;
		onClickCallback = null;
		handlePanelClickCallback = null;
	});

	describe('Panel basic behavior', () => {
		it('exists and renders with mock props', () => {
			expect(panel).toMatchSnapshot();
		});

		it(`has the class ${ACTIVEPANEL_CLASS} when props is open`, function() {
			const node = openPanel.find(`.${ACTIVEPANEL_CLASS}`);
			expect(node.length).toBe(1);
		});

		it('calls handlePanelClick callback onClick', function() {
			const accPanel = panel.find('[aria-controls="panel-firstsection"]');
			const isOpen = panel.prop('isOpen');
			const mockClick = {
				target: accPanel,
				preventDefault: jest.fn(),
				stopPropagation: jest.fn(),
			};

			expect(handlePanelClickCallback).not.toHaveBeenCalled();
			accPanel.simulate('click', mockClick);
			expect(handlePanelClickCallback).toHaveBeenCalledWith(mockClick, {
				panelIndex: panel.prop('panelIndex'),
				isOpen: !isOpen,
			});
		});

		it('calls setPanelState callback onKeyUp', function() {
			const accPanel = panel.find('[aria-controls="panel-firstsection"]');
			const isOpen = panel.prop('isOpen');
			const mockKeyUp = { key: 'Enter', preventDefault: jest.fn() };

			expect(handlePanelClickCallback).not.toHaveBeenCalled();
			accPanel.simulate('keyUp', mockKeyUp);
			expect(handlePanelClickCallback).toHaveBeenCalledWith(mockKeyUp, {
				panelIndex: panel.prop('panelIndex'),
				isOpen: !isOpen,
			});
		});

		it('calls onClickCallback onClick', function() {
			const accPanel = panel.find('[aria-controls="panel-firstsection"]');
			const isOpen = panel.prop('isOpen');
			const fakeEvent = {
				target: panel,
				preventDefault: jest.fn(),
				stopPropagation: jest.fn(),
			};

			expect(onClickCallback).not.toHaveBeenCalled();
			accPanel.simulate('click', fakeEvent);
			expect(onClickCallback).toHaveBeenCalledWith(fakeEvent, !isOpen);
		});

		it('renders an icon component', function() {
			const node = panel.find(Icon);
			expect(node.length).toBe(1);
		});
	});

	describe('Panel with right aligned icon', () => {
		let panelLeftIcon;

		beforeEach(() => {
			panelLeftIcon = mount(
				<AccordionPanel
					label="First Section"
					indicatorAlign="left"
					panelContent={
						<div className="runningText">
							<p>{textContent1}</p>
						</div>
					}
				/>
			);
		});

		afterEach(() => {
			panelLeftIcon = null;
		});

		it('exists and renders icon left', () => {
			expect(panelLeftIcon).toMatchSnapshot();
		});

		it('reverses Flex direction when icon is right-aligned', () => {
			const node = panelLeftIcon.find('.atAll_flex--rowReverse');
			expect(node.length).toBe(1);
		});
	});

	describe('Panel with custom icon', () => {
		let panelCustomIcon;

		beforeEach(() => {
			panelCustomIcon = shallow(
				<AccordionPanel
					label="First Section"
					indicatorAlign="left"
					indicatorIcon={customIcon}
					indicatorIconActive={customIconActive}
					panelContent={
						<div className="runningText">
							<p>{textContent1}</p>
						</div>
					}
				/>
			);
		});

		afterEach(() => {
			panelCustomIcon = null;
		});

		it('exists and renders a custom icon', function() {
			expect(panelCustomIcon).toMatchSnapshot();
		});
	});

	describe('Panel with toggle switch', () => {
		let panelSwitch, handleToggleSpy;

		beforeEach(() => {
			handleToggleSpy = jest
				.spyOn(AccordionPanel.prototype, '_handleToggle')
				.mockImplementation(() => {});

			panelSwitch = mount(
				<AccordionPanel
					indicatorSwitch
					label="First Section"
					indicatorAlign="left"
					panelContent={
						<div className="runningText">
							<p>{textContent1}</p>
						</div>
					}
				/>
			);
		});

		afterEach(() => {
			panelSwitch = null;
			handleToggleSpy.mockRestore();
		});

		it('exists and renders a toggle switch', function() {
			expect(panelSwitch).toMatchSnapshot();
		});

		it('calls handleToggle when toggle switch is clicked', function() {
			const switchNode = panelSwitch.find(ToggleSwitch);
			expect(handleToggleSpy).not.toHaveBeenCalled();
			switchNode.find('button').simulate('click');
			expect(handleToggleSpy).toHaveBeenCalled();
		});

		it('calls custom function and does not call handleToggle when toggle switch is clicked and onToggleClick is specified', function() {
			const onToggleClickCallback = jest.fn();
			const onToggleClickSpy = jest.spyOn(
				AccordionPanel.prototype,
				'onToggleClick'
			);
			const panelOnToggleClick = mount(
				<AccordionPanel
					indicatorSwitch
					onToggleClick={onToggleClickCallback}
					label="First Section"
					indicatorAlign="left"
					panelContent={
						<div className="runningText">
							<p>{textContent1}</p>
						</div>
					}
				/>
			);
			const switchNode = panelOnToggleClick.find(ToggleSwitch);
			expect(onToggleClickSpy).not.toHaveBeenCalled();
			switchNode.find('button').simulate('click');
			expect(handleToggleSpy).not.toHaveBeenCalled();
			expect(onToggleClickSpy).toHaveBeenCalled();
		});
		it('ToggleSwitch should be disabled if disableAndOpen params passed', () => {
			const panelOnToggleClick = mount(
				<AccordionPanel
					indicatorSwitch
					disableAndOpen
					label="First Section"
					indicatorAlign="left"
					panelContent={
						<div className="runningText">
							<p>{textContent1}</p>
						</div>
					}
				/>
			);
			const switchNode = panelOnToggleClick.find(ToggleSwitch);
			expect(switchNode.props().disabled).toBe(true);
		});
	});

	describe('Locked panel with toggle switch', () => {
		let panelToggleSwitchLocked;
		const onLockedLabelClickMock = jest.fn();
		const onClickCallbackMock = jest.fn();

		beforeEach(() => {
			panelToggleSwitchLocked = mount(
				<AccordionPanel
					indicatorSwitch
					label="First Section"
					indicatorAlign="left"
					panelContent={
						<div className="runningText">
							<p>{textContent1}</p>
						</div>
					}
					isLocked
					lockedLabel="Unlock me!"
					onLockedLabelClick={onLockedLabelClickMock}
					onClickCallback={onClickCallbackMock}
				/>
			);
		});

		afterEach(() => {
			panelToggleSwitchLocked = null;
			onLockedLabelClickMock.mockClear();
		});

		it('renders locked toggle panel', () => {
			expect(panelToggleSwitchLocked).toMatchSnapshot();
		});

		it('should call onLockedLabelClick callback when user clicks on locked label', () => {
			const lockedLabel = panelToggleSwitchLocked.find('.lockedBadge-badge');
			lockedLabel.simulate('click');
			expect(onLockedLabelClickMock).toHaveBeenCalledTimes(1);
		});

		it('shouln`t call onClickCallback when user clicks and panel is locked', () => {
			const switchNode = panelToggleSwitchLocked.find(ToggleSwitch);
			switchNode.find('button').simulate('click');
			expect(onClickCallback).toHaveBeenCalledTimes(0);
		});
	});
});
