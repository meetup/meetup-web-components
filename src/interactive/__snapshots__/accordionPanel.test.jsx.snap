// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionPanel Panel basic behavior exists and renders with mock props 1`] = `
<Flex
  className="accordionPanel"
  direction="row"
  rowReverse={false}
>
  <FlexItem>
    <button
      aria-controls="panel-firstsection"
      aria-expanded={false}
      aria-selected={false}
      className="accordionPanel-label display--block span--100 padding--bottom"
      id="label-firstsection"
      onClick={[Function]}
      role="tab"
    >
      First Section
    </button>
    <Chunk
      aria-hidden={true}
      aria-labelledby="label-firstsection"
      className="accordionPanel-animator accordionPanel-animator--collapse"
      onTransitionEnd={[Function]}
      role="tabpanel"
      style={
        Object {
          "height": "0px",
        }
      }
    >
      <div
        className="accordionPanel-content"
      >
        <div
          className="runningText"
        >
          <p>
            Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
          </p>
        </div>
      </div>
    </Chunk>
  </FlexItem>
  <FlexItem
    className="accordionPanel-icon"
    onClick={[Function]}
    shrink={true}
  >
    <Icon
      shape="chevron-down"
      size="xs"
    />
  </FlexItem>
</Flex>
`;

exports[`AccordionPanel Panel with custom icon exists and renders a custom icon 1`] = `
<Flex
  className="accordionPanel"
  direction="row"
  rowReverse="all"
>
  <FlexItem>
    <button
      aria-controls="panel-firstsection"
      aria-expanded={false}
      aria-selected={false}
      className="accordionPanel-label display--block span--100 padding--bottom"
      id="label-firstsection"
      onClick={[Function]}
      role="tab"
    >
      First Section
    </button>
    <Chunk
      aria-hidden={true}
      aria-labelledby="label-firstsection"
      className="accordionPanel-animator accordionPanel-animator--collapse"
      onTransitionEnd={[Function]}
      role="tabpanel"
      style={
        Object {
          "height": "0px",
        }
      }
    >
      <div
        className="accordionPanel-content"
      >
        <div
          className="runningText"
        >
          <p>
            Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
          </p>
        </div>
      </div>
    </Chunk>
  </FlexItem>
  <FlexItem
    className="accordionPanel-icon"
    onClick={[Function]}
    shrink={true}
  >
    <Icon
      shape="plus"
      size="xs"
    />
  </FlexItem>
</Flex>
`;

exports[`AccordionPanel Panel with right aligned icon exists and renders icon left 1`] = `
<AccordionPanel
  indicatorAlign="left"
  indicatorIcon="chevron-down"
  indicatorIconSize="xs"
  isOpen={false}
  label="First Section"
  panelContent={
    <div
      className="runningText"
    >
      <p>
        Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
      </p>
    </div>
  }
>
  <Flex
    className="accordionPanel"
    direction="row"
    rowReverse="all"
  >
    <div
      className="flex flex--row atAll_flex--rowReverse accordionPanel"
    >
      <FlexItem>
        <div
          className="flex-item"
        >
          <button
            aria-controls="panel-firstsection"
            aria-expanded={false}
            aria-selected={false}
            className="accordionPanel-label display--block span--100 padding--bottom"
            id="label-firstsection"
            onClick={[Function]}
            role="tab"
          >
            First Section
          </button>
          <Chunk
            aria-hidden={true}
            aria-labelledby="label-firstsection"
            className="accordionPanel-animator accordionPanel-animator--collapse"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "0px",
              }
            }
          >
            <div
              aria-hidden={true}
              aria-labelledby="label-firstsection"
              className="chunk accordionPanel-animator accordionPanel-animator--collapse"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "0px",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div
                  className="runningText"
                >
                  <p>
                    Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                  </p>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </FlexItem>
      <FlexItem
        className="accordionPanel-icon"
        onClick={[Function]}
        shrink={true}
      >
        <div
          className="flex-item flex-item--shrink accordionPanel-icon"
          onClick={[Function]}
        >
          <Icon
            shape="chevron-down"
            size="xs"
          >
            <span
              className="svg svg--chevron-down"
            >
              <svg
                className="svg-icon valign--middle"
                height="16"
                preserveAspectRatio="xMinYMin meet"
                role="img"
                style={Object {}}
                viewBox="0 0 16 16"
                width="16"
              >
                <use
                  xlinkHref="#icon-chevron-down--small"
                />
              </svg>
            </span>
          </Icon>
        </div>
      </FlexItem>
    </div>
  </Flex>
</AccordionPanel>
`;

exports[`AccordionPanel Panel with toggle switch exists and renders a toggle switch 1`] = `
<AccordionPanel
  indicatorAlign="left"
  indicatorIcon="chevron-down"
  indicatorIconSize="xs"
  indicatorSwitch={true}
  isOpen={false}
  label="First Section"
  panelContent={
    <div
      className="runningText"
    >
      <p>
        Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
      </p>
    </div>
  }
>
  <Flex
    className="accordionPanel"
    direction="row"
    rowReverse="all"
  >
    <div
      className="flex flex--row atAll_flex--rowReverse accordionPanel"
    >
      <FlexItem>
        <div
          className="flex-item"
        >
          <button
            aria-controls="panel-firstsection"
            aria-expanded={false}
            aria-selected={false}
            className="accordionPanel-label display--block span--100 padding--bottom"
            id="label-firstsection"
            onClick={[Function]}
            role="tab"
          >
            First Section
          </button>
          <Chunk
            aria-hidden={true}
            aria-labelledby="label-firstsection"
            className="accordionPanel-animator accordionPanel-animator--collapse"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "0px",
              }
            }
          >
            <div
              aria-hidden={true}
              aria-labelledby="label-firstsection"
              className="chunk accordionPanel-animator accordionPanel-animator--collapse"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "0px",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div
                  className="runningText"
                >
                  <p>
                    Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                  </p>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </FlexItem>
      <FlexItem
        className="accordionPanel-icon"
        onClick={[Function]}
        shrink={true}
      >
        <div
          className="flex-item flex-item--shrink accordionPanel-icon"
          onClick={[Function]}
        >
          <WithToggle
            disabled={false}
            id="firstsection-switch"
            isActive={false}
            name="firstsection"
            onClick={[Function]}
          >
            <span
              aria-pressed={false}
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              <ToggleSwitch
                disabled={false}
                id="firstsection-switch"
                isActive={false}
                name="firstsection"
                onClick={[Function]}
                toggleActive={[Function]}
              >
                <div
                  id="firstsection-switch"
                  onClick={[Function]}
                >
                  <Button
                    aria-checked={false}
                    aria-labelledby="firstsection-label"
                    className="toggleSwitch"
                    disabled={false}
                    onClick={[Function]}
                    reset={true}
                    role="switch"
                    tabIndex={-1}
                  >
                    <button
                      aria-checked={false}
                      aria-labelledby="firstsection-label"
                      className="button button--reset toggleSwitch"
                      onClick={[Function]}
                      role="switch"
                      tabIndex={-1}
                    >
                      <span
                        className="toggleSwitch-knob flex flex--center flex--alignCenter"
                      >
                        <Icon
                          color="rgba(46,62,72,0.6)"
                          label="Toggle switch label"
                          shape="cross"
                          size="xxs"
                        >
                          <span
                            className="svg svg--cross"
                          >
                            <svg
                              className="svg-icon valign--middle"
                              height="12"
                              label="Toggle switch label"
                              preserveAspectRatio="xMinYMin meet"
                              role="img"
                              style={
                                Object {
                                  "fill": "rgba(46,62,72,0.6)",
                                }
                              }
                              viewBox="0 0 12 12"
                              width="12"
                            >
                              <use
                                xlinkHref="#icon-cross--small"
                              />
                            </svg>
                          </span>
                        </Icon>
                      </span>
                    </button>
                  </Button>
                </div>
              </ToggleSwitch>
            </span>
          </WithToggle>
        </div>
      </FlexItem>
    </div>
  </Flex>
</AccordionPanel>
`;
